### ISSUE TITLE
"Error while communicating with YubiKey: authentication error" when using `--pin-policy always`

### ISSUE THREAD

**Issue by Ovyerus** [str4d/age-plugin-yubikey#80]
## Environment

* OS: Windows 10, macOS Monterey
* age-plugin-yubikey version: 0.3.0
* Yubikey 5C NFC

## What were you trying to do

Trying to generate a new Age key through `age-plugin-yubikey` using the `--pin-policy always` option, but running into a weird error which causes it to fail generation after touching the YubiKey, whereas omitting this option or setting it to `never` succeeds as expected.

Fiddling with the option options like `--touch-policy` does not cause this problem.

## What happened

```
$ age-plugin-yubikey --generate --pin-policy always

Enter PIN for YubiKey with serial 22317870(default is 123456): [hidden]
👆 Please touch the YubiKey
Error: Error while communicating with YubiKey: authentication error

[ Did this not do what you expected? Could an error be more useful? ]
[ Tell us: https://str4d.xyz/age-plugin-yubikey/report              ]
```

Omitting `--pin-policy always` succeeds as normal
```
$ age-plugin-yubikey --generate

Enter PIN for YubiKey with serial 22317870 (default is 123456): [hidden]
👆 Please touch the YubiKey
#       Serial: 22317870, Slot: 1
#         Name: age identity c8caa1bd
#      Created: Fri, 04 Nov 2022 12:12:26 +0000
#   PIN policy: Once   (A PIN is required once per session, if set)
# Touch policy: Always (A physical touch is required for every decryption)
#    Recipient: age1yubikey1qdszmlws3r8f69argu7u2kkxp02nnfcdsxmjfs49ghehc6ufp6dqjxw0lyk
AGE-PLUGIN-YUBIKEY-19694GQVZER92R0GZMVQ69
```

**Comment by frobiac** [comment]
I get the same error message when trying to generate a key on a Yubikey NFC 5 with USB-C connector (VoidLinux, also plugin version 0.3.0).

But trying all possible combinations of options mentioned above did not allow key generation in my case. I also tried switching the management key from default to generated and back, which was something I had noted down when I was able to generate a new key with a second NFC 5 YubiKey but with USB-A connector without any issues a few weeks ago.

All other functionality of the 5C version with other tools seems to work fine, so I doubt it is a general HW or OS configuration issue.

Both YubiKeys are on 5.4.3 firmware and have all the same options and apps enabled, as far as I can tell.

**Comment by str4d** [comment]
Hmm, I'm not sure what is going on here. Something we had a problem with in the past was that if the user needed to enter a PIN but touched the YubiKey too early, it would "type" an OTP as the PIN. We now parse the PIN to detect this case and re-ask for a PIN (#46), but maybe something similar is happening here with some other YubiKey applet, or maybe that fix has regressed.

**Comment by str4d** [comment]
@Ovyerus are you still able to reproduce this issue with `age 0.3.0`? And if so, could you please also try using the current `main` branch and check if the problem is reproducible there?

**Comment by Ovyerus** [comment]
Still reproducible on `v0.3.3`, seems to be fixed on the current main branch commit (22fd00bd22ee8a8fe9f3b33db2e4b7eea944bf37)!

```
▲ cargo run -- --generate --pin-policy always
    Finished dev [unoptimized + debuginfo] target(s) in 0.23s
     Running `target/debug/age-plugin-yubikey --generate --pin-policy always`
🎲 Generating key...

Enter PIN for YubiKey with serial 22317870 (default is 123456): [hidden]

🔏 Generating certificate...
Enter PIN for YubiKey with serial 22317870: [hidden]
👆 Please touch the YubiKey
#       Serial: 22317870, Slot: 2
#         Name: age identity ef4d2e57
#      Created: Thu, 23 Mar 2023 11:42:06 +0000
#   PIN policy: Always (A PIN is required for every decryption, if set)
# Touch policy: Always (A physical touch is required for every decryption)
#    Recipient: age1yubikey1qwnnd7277p43ygzaq7qf86mzmax2j52qfg2f98m5fyy3zl7ng6fvcjjprha
AGE-PLUGIN-YUBIKEY-19694GQVRAAXJU4CY0Z6CP
```

**Comment by str4d** [comment]
Excellent! In that case I'll close this as fixed in v0.4.0, which I'll be releasing soon 🙂
